developers/blobstream-rollups.md

```diff
@@ -0,0 +1,399 @@
+
+The second difference is the proof types.
+In the case of a fraud proof,
+the proofs outlined in the [proofs details](#proof-details) section would need
+to be verified to be sure that the share containing the invalid state transition is part of the rollup data.
+Alternatively, the rollup settlement contract would need to have a library to parse protobuf encoded PFBs,
+as explained in the [compact proofs](#compact-proofs) section, to have less expensive proofs.
+The cost of parsing the protobuf is not included in this analysis and needs to be investigated separately.
+
+#### Pros
+
+- Using the same share commitment as the one saved in Celestia which gives access to existing tooling
+
+#### Cons
+
+- The proofs are expensive in the base case.
+  And if the settlement contract is able to parse the PFBs,
+  thorough investigations of the cost of that would need to be done.
+
+## Zk-rollups
+
+Zk-rollups, aka validity rollups, can also use Celestia as a DA and Blobstream to verify that the data was posted.
+However, the submission process is different from the above constructions,
+since there are no fraud proofs,
+and everything should be verified when submitting the commitment to the settlement contract.
+
+Similar to the optimistic case,
+the rollup settlement contract can reference the rollup data
+using either the sequence of spans approach or the share commitments.
+We will discuss both in this section.
+
+### Zk-rollups that use sequence of spans
+
+When submitting the commitments to the rollup settlement contract,
+this latter will need to verify the following:
+
+1. Zk-proof of the state transitions, which is left for the rollup to define.
+2. Verify that the sequence of spans is [valid](https://github.com/celestiaorg/blobstream-contracts/blob/master/docs/inclusion-proofs.md),
+   i.e., is part of the Celestia block referenced by its height,
+   as described in the [proof details](#proof-details-1) section.
+3. Zk-proof of the rollup data to the data root.
+   The verification process of this should accept a commitment as input
+   so that the settlement contract makes sure it's the correct value that's being saved.
+   The commitment can be the data root and the sequence of spans.
+   And, when the rollup data is proven inside the circuit to the data root,
+   the used data root is asserted to be the input one.
+   Similarly, the data's location is asserted to be the same as the input sequence of spans.
+   These arguments are the ones used in the sequence of spans verification in (2).
+
+Once these are valid, the settlement contract can be sure that the rollup data was posted to Celestia,
+and the sequence of spans references it correctly.
+
+#### Pros
+
+- The inclusion proof inside the zk-circuit is a simple proof that uses traditional merkle tree.
+  In the case of using share commitment, as will be explained below,
+  additional libraries that can be expensive to prove are required.
+
+#### Cons
+
+None
```

Comment:
same question as before, if there are no cons, then can we remove other sections? if not, then we should add cons here